scribble.go

```diff
@@ -199,6 +199,39 @@ func (d *Driver) ReadAll(collection string) ([]string, error) {
 	return records, nil
 }
 
+// List all records from a collection; this is returned as a slice of strings because
+// there is no way of knowing what type the record is.
+func (d *Driver) List(collection string) ([]string, error) {
+
+	// ensure there is a collection to read
+	if collection == "" {
+		return nil, fmt.Errorf("missing collection - unable to record location")
+	}
+
+	//
+	dir := filepath.Join(d.dir, collection)
+
+	// check to see if collection (directory) exists
+	if _, err := stat(dir); err != nil {
+		return nil, nil
+	}
+
+	// read all the files in the transaction.Collection; an error here just means
+	// the collection is either empty or doesn't exist
+	files, _ := ioutil.ReadDir(dir)
+
+	// the IDs of collection
+	var records []string
+
+	for _, file := range files {
+		name := file.Name()
```

Comment:
Breaks current standard of including the absolute path.

Reply:
In the same way you do a list of files in the ReadAll() function, so I'm not sure what standard you are talking about.

In the List() function, I don't see the need to set an absolute path to make a list of names.
I have the impression that you took a quick look at this code or I did not clearly describe what it does.